app/Http/Controllers/ProductController.php

```diff
@@ -0,0 +1,225 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Group;
+use App\Models\Product;
+use App\Models\ProductProductSize;
+use App\Models\ProductSize;
+use App\Models\ProductType;
+use App\Models\OrderLine;
+use App\Http\Requests\ProductCreationRequest;
+use App\Http\Requests\ProductEditRequest;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Storage;
+
+class ProductController extends Controller
+{
+    public function index()
+    {
+        $products = Product::with('productType', 'groups')->get();
+        $categories = ProductType::all();
+        $productsModel = $products->map(function ($product) {
+            return [
+                'name' => $product->name,
+                'category' => $product->productType->type,
+                'groups' => $product->groups->pluck('name')->toArray(),
+                'sizesWithPrices' => $this->getSizesWithPrices($product),
+                'id' => $product->id,
+            ];
+        });
+
+        return view('admin.products', ['products' => $productsModel, 'categories' => $categories]);
+    }
+
+    private function getSizesWithPrices($product)
+    {
+        $sizesWithPrices = ProductProductSize::where('product_id', $product->id)->get();
+        $sizes = [];
+        foreach ($sizesWithPrices as $sizeWithPrice) {
```

Comment:
Deze foreach is volgensmij niet nodig omdat je $product->productSizes()->first(); kan doen